src/pages/onboarding/ChoosePlan.js

```
@@ -0,0 +1,115 @@
+import React, { Component, Fragment } from 'react';
+import { reduxForm, formValueSelector } from 'redux-form';
+import { connect } from 'react-redux';
+import { Panel, Grid, Button } from '@sparkpost/matchbox';
+import { showAlert } from 'src/actions/globalAlert';
+import { CenteredLogo, Loading, PlanPicker } from 'src/components';
+import Steps from './components/Steps';
+import { getPlans } from 'src/actions/account';
+import { getBillingCountries, billingCreate } from 'src/actions/billing';
+import { publicPlansSelector } from 'src/selectors/accountBillingInfo';
+import { onboardingInitialValues } from 'src/selectors/accountBillingForms';
+import PaymentForm from 'src/pages/billing/forms/fields/PaymentForm';
+import BillingAddressForm from 'src/pages/billing/forms/fields/BillingAddressForm';
+
+const FORM_NAME = 'selectInitialPlan';
+
+export class OnboardingPlanPage extends Component {
+  componentDidMount() {
+    this.props.getPlans();
+    this.props.getBillingCountries();
```

Do we need to `.catch` these?

---

And perhaps even redirect to the dashboard or the next step if these fail? The form doesn't load properly. I'm considering ApiErrorBanner, but redirecting might be the better experience.

---

I thought we were holistically going to solve the uncaught requests from cDM

---

Is that the plan? I'm out of the loop with whats planned there

---

These are two different issues:
1) Catching/swallowing API errors so they don't cause uncaught promise rejections when we've already handled the error via _FAIL actions (should solve holistically)
2) Dealing with individual error cases as needed

We should probably be dealing with (2) as we encounter it, so we don't add to the problem while we are waiting to holistically solve (1) -- which means _not_ using `.catch()` here but handling plan/country errors one way or another with a redirect like @jonambas suggests or something similar.

---

handling in cDU and redirect to next step